hnn_core/optimization/general_optimization.py

```diff
@@ -0,0 +1,407 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from .metrics import _rmse_evoked
+
+from skopt import gp_minimize
```

Comment:
you should nest this import and make `'cobyla'` the default. I'm hesitant to add another hard dependency. It will make our life difficult for installation